lib/pliny/middleware/rescue_errors.rb

```diff
@@ -6,14 +6,20 @@ def initialize(app, options = {})
     end
 
     def call(env)
+      error = nil
       @app.call(env)
     rescue Pliny::Errors::Error => e
+      error = e
       Pliny::Errors::Error.render(e)
     rescue => e
+      error = e
       raise if @raise
 
       Pliny::ErrorReporters.notify(e, rack_env: env)
       Pliny::Errors::Error.render(Pliny::Errors::InternalServerError.new)
+    ensure
+      # Leave this in env for CanonicalLogLine.
+      env["pliny.error"] = e
```

Comment:
Yeah, good point, it's a little messy. I refactored a bit in 2e384c3.

I'm a little hesitant to use an `attr_reader` because a middleware's exposed interface `call(env)` is pretty suggestive that the instance itself should be stateless throughout any single request (i.e. you could potentially run two requests through one instance at the same time).